src/scheduler/simulate.c

```diff
@@ -901,18 +911,22 @@ create_events(server_info *sinfo)
 		if (node->is_sleeping) {
 			te = create_event(TIMED_NODE_UP_EVENT, sinfo->server_time + PROVISION_DURATION,
 					(event_ptr_t *) node, (event_func_t) node_up_event, NULL);
-			if (te == NULL)
+			if (te == NULL) {
+				free(all_resresv_copy);
 				return 0;
+			}
```

Comment:
I think this is actually a bug.  If we return here, we'll leak events.  Instead of returning, errflag++ and break (that is what happens above).  That will cause the code below to free everything and return.